### PR TITLE
kmc-solid: `#![forbid(unsafe_op_in_unsafe_fn)]`

### PR DESCRIPTION
**Pull request by workingjubilee**
The path logic _should_ handle the forbiddance in the itron sources correctly, despite them being an "out-of-line" module.

Part of https://github.com/rust-lang/rust/issues/127747

### CODE REVIEW

**Comment by workingjubilee** [comment]
cc @kawadakk
r? libs

**Review by kawadakk — APPROVED** [review]

File: library/std/src/sys/pal/itron/abi.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 //! ABI for μITRON derivatives
+#![forbid(unsafe_op_in_unsafe_fn)]
```
Comment (Amanieu) [inline library/std/src/sys/pal/itron/abi.rs:2]: Everything under itron/ is already a sub-module of library/std/src/os/solid/mod.rs. so this should already be covered by the forbid above. The directory layout is a bit strange.

Comment (workingjubilee) [reply library/std/src/sys/pal/itron/abi.rs:2]: Ah is that how that works? Okay!

File: library/std/src/os/solid/io.rs [hunk 2]
```diff
@@ -44,7 +44,7 @@
 //!
 //! [`BorrowedFd<'a>`]: crate::os::solid::io::BorrowedFd
 
-#![deny(unsafe_op_in_unsafe_fn)]
+#![forbid(unsafe_op_in_unsafe_fn)]
```
[the remark this answers is not in the document]
Comment (workingjubilee) [reply library/std/src/os/solid/io.rs:47]: oh true!

**Pull request merged**